src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -41,16 +41,29 @@ internal static partial class PathInternal
         /// </summary>
         internal static bool IsPathTooLong(string fullPath)
         {
+            // We'll never know precisely what will fail as paths get changed internally in Windows and
+            // may grow to exceed MaxExtendedPath. We'll only try to catch ones we know will absolutely
+            // fail.
+
+            if (fullPath.Length < MaxExtendedPath - UncExtendedPathPrefix.Length)
+            {
+                // For sure is ok
```

Comment:
For our purposes it's ok.  We won't put it over the max by prepending.  I'll clarify.